### PR TITLE
Remove pymoab as a Dependency

### PR DESCRIPTION
**Pull request by ahnaf-tahmid-chowdhury**
## Description

This PR removes `pymoab` from `pyproject.toml` as a dependency.

## Changes

Here are the main changes made in this PR:

1. **`pyproject.toml`:** Removed `pymoab` from the list of dependencies.

2. **`dagnav.py`:** Added instructions to handle the case if `pymoab` is not found.

### CODE REVIEW

File: dagmc/dagnav.py
Comment (pshriwise) [inline dagmc/dagnav.py:6]:
```suggestion
modeling and simulation built upon the PyMOAB package.
```

File: dagmc/dagnav.py
Comment (pshriwise) [inline dagmc/dagnav.py:20]: I know I suggested this in #1, but the message is a overly verbose for my taste. There are (somewhat dated) instructions on how to build MOAB with PyMOAB enabled [here](https://ftp.mcs.anl.gov/pub/fathom/moab-docs/building.html). I think we should point to them in this message and update the installation instructions there.

Also, while we may want to add a logging capability to this project in the future, I don't see the point of importing that module for this single message. We can raise our own message from a `ModuleNotFoundError` 

```python

try:
    from pymoab import core, types, rng, tag
except ImportError as e:
    msg = 'PyMOAB package was not found, please install this package using the instructions found here: https://ftp.mcs.anl.gov/pub/fathom/moab-docs/building.html'
    raise ModuleNotFoundError(msg) from e
```

**Review by pshriwise — CHANGES_REQUESTED** [review]
A couple of comments form me, but I think this is close!

**Review by gonuke — APPROVED** [review]
LGTM - I'll let @pshriwise decide if it's ready

**Review by pshriwise — APPROVED** [review]

**Pull request merged**
